console-subscriber/src/builder.rs

```diff
@@ -501,8 +520,21 @@ impl Builder {
                     .enable_time()
                     .build()
                     .expect("console subscriber runtime initialization failed");
-
                 runtime.block_on(async move {
+                    #[cfg(feature = "grpc-web")]
+                    if enable_grpc_web {
+                        server
+                            .serve_with_grpc_web()
+                            .await
+                            .expect("console subscriber server failed")
+                    } else {
+                        server
+                            .serve()
+                            .await
+                            .expect("console subscriber server failed")
+                    }
+
+                    #[cfg(not(feature = "grpc-web"))]
                     server
                         .serve()
                         .await
```

Comment:
Having the `else` implementation effectively twice isn't ideal. I wonder if there's a better way to do this.

If we only need to modify the transport (based on comments in `serve_with_grpc`), then perhaps something like this would work:


```suggestion
                    let transport = tonic::transport::Server::default();
                    
                    #[cfg(feature = "grpc-web")]
                    let transport = transport.accept_http1(self.enable_grpc_web);
                    
                    server
                        .serve_with(transport)
                        .await
```

Note, I haven't actually tried to compile this code.